tests/ui/lint/invalid_null_args.stderr

```diff
@@ -0,0 +1,289 @@
+  --> $DIR/invalid_null_args.rs:25:23
+   |
+LL |     let _: &[usize] = std::slice::from_raw_parts(0 as *mut _, 0);
+   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/std/ptr/index.html> and <https://doc.rust-lang.org/reference/behavior-considered-undefined.html>
+note: null pointer originates from here
+  --> $DIR/invalid_null_args.rs:25:50
+   |
+LL |     let _: &[usize] = std::slice::from_raw_parts(0 as *mut _, 0);
+   |                                                  ^
+
+error: calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
+  --> $DIR/invalid_null_args.rs:27:23
+   |
+LL |     let _: &[usize] = std::slice::from_raw_parts(mem::transmute(0usize), 0);
+   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/std/ptr/index.html> and <https://doc.rust-lang.org/reference/behavior-considered-undefined.html>
+note: null pointer originates from here
+  --> $DIR/invalid_null_args.rs:27:65
+   |
+LL |     let _: &[usize] = std::slice::from_raw_parts(mem::transmute(0usize), 0);
+   |                                                                 ^^^^^^
+
+error: calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
+  --> $DIR/invalid_null_args.rs:30:23
+   |
+LL |     let _: &[usize] = std::slice::from_raw_parts_mut(ptr::null_mut(), 0);
+   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/std/ptr/index.html> and <https://doc.rust-lang.org/reference/behavior-considered-undefined.html>
+
+error: calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
+  --> $DIR/invalid_null_args.rs:33:5
+   |
+LL |     ptr::copy::<usize>(ptr::null(), ptr::NonNull::dangling().as_ptr(), 0);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/std/ptr/index.html> and <https://doc.rust-lang.org/reference/behavior-considered-undefined.html>
+
+error: calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
+  --> $DIR/invalid_null_args.rs:35:5
+   |
+LL |     ptr::copy::<usize>(ptr::NonNull::dangling().as_ptr(), ptr::null_mut(), 0);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/std/ptr/index.html> and <https://doc.rust-lang.org/reference/behavior-considered-undefined.html>
+
+error: calling this function with a null pointer is Undefined Behavior, even if the result of the function is unused
```

Comment:
```suggestion
error: it is undefined behavior to call this function with a null pointer, even if the result of the function is unused
```

We already have a diagnostic that says "it is undefined behavior to use this value", so making this follow that format feels nicer to me. Also we don't capitalize UB in diagnostics so we should follow that: https://github.com/search?q=repo%3Arust-lang%2Frust+undefined+behavior+path%3Atests%2Fui&type=code

Reply:
We also [have](https://github.com/rust-lang/rust/blob/45b40a75966b36d3588f173441896fddad01cd80/tests/ui/lint/reference_casting.stderr) [diagnostics](https://github.com/rust-lang/rust/blob/45b40a75966b36d3588f173441896fddad01cd80/tests/ui/lint/invalid_from_utf8.stderr) of the form I used.

As for `Undefined Behavior` being capitalized I took this style from miri in [their diagnostics](https://github.com/search?q=repo%3Arust-lang%2Frust+undefined+behavior+path%3Asrc%2Ftools%2Fmiri%2Ftests&type=code).

Reply:
Keeping the form you used is fine, I'd still want it to not be capitalized. This is a rustc lint so it should follow what rustc uses. (We really should have a large whole-project discussion on this though, just to see if we may find a consensus)

Reply:
I think I convinced RalfJ that "undefined behavior" should not be capitalized (just as we don't capitalize "race condition") -- and he later [updated the Reference](https://github.com/rust-lang/reference/pull/1575) accordingly -- but prior to that, he had capitalized it in a lot of places (like Miri), and those haven't all been updated yet.

As a stylistic matter in the Reference, we write "undefined behavior".  I push the text in lang RFCs this direction as well.